geo-traits/src/point.rs

```diff
@@ -34,11 +34,11 @@ impl<T: CoordNum> PointTrait for Point<T> {
 }
 
 #[cfg(feature = "geo-types")]
-impl<T: CoordNum> PointTrait for &Point<T> {
-    type CoordType<'a>
+impl<'a, T: CoordNum> PointTrait for &'a Point<T> {
+    type CoordType<'b>
         = &'a Coord<<Self as GeometryTrait>::T>
```

Comment:
Ah! This is the line that I missed. The `&'a` here used to refer to the `CoordType<'a>`, but now references the lifetime of the `impl &'a Point`

I see now how this is now more relaxed. Thanks for the test!